scripts/clean_up.py

```diff
@@ -0,0 +1,121 @@
+""" This script cleans up the sequence data and metadata within GCS and Metamist """
+
+import logging
+import coloredlogs
+import click
+
+from cloudpathlib import CloudPath
+from google.cloud import storage
+
+from sample_metadata.apis import SampleApi, SequenceApi
+from sample_metadata.models import (
+    SequenceUpdateModel,
+)
+
+
+client = storage.Client()
+sapi = SampleApi()
+seqapi = SequenceApi()
+coloredlogs.install()
+
+
+def clean_up_metamist(
+    sequence_ids_internal: list[str],
+    clean_up_location: str,
+):
+    """Deleting sequencing entires for specified sequences in metamist
+    This will delete all of the reads for the latest sequence."""
+    all_locations: list[CloudPath] = []
+    sequence_ids: list[int] = []
+    for sequence_id in sequence_ids_internal:
+        current_sequence = seqapi.get_sequence_by_id(sequence_id)
+        reads = current_sequence['meta']['reads']
+        current_read_locations: list[CloudPath] = []
+        # Skip delete operations if there are no reads for the given sequence
+        delete_reads = bool(reads)
+        for read in reads:
+            for strand in read:
+                cloud_path = strand['location']
+                read_blob = CloudPath(cloud_path)
+                blob_location = read_blob.anchor + read_blob.drive
+                # Validates the file that we want to delete, is in the bucket specified.
+                if read_blob.exists() and blob_location == clean_up_location:
+                    current_read_locations.append(read_blob)
+                else:
+                    logging.error(
+                        f'{read_blob.name} was not found in {clean_up_location}. Skipping delete of {sequence_id} from metamist'
+                    )
+                    # Skip delete operations, something might be wrong with the reads meta locations.
+                    delete_reads = False
+                    continue
+        # Clear reads metadata, including location, basename, class, checksum & size
+        if delete_reads:
```

Comment:
You'll want to clear the buckets before the sequence update, to ensure we don't fail between that process.

Reply:
have moved things around accordingly